### ISSUE TITLE
permit interoperability of ssm device with ssam

### ISSUE THREAD

**Issue by namdre** [eclipse/sumo#8051]
https://highways.dot.gov/research/safety/ssam/surrogate-safety-assessment-model-overview

**Comment by m-kro** [comment]
To my understanding, SSM device and SSAM implementation both calculate surrogate safety measures. If any, vehicle data like from [FCD output](https://sumo.dlr.de/docs/Simulation/Output/FCDOutput.html) and SUMO vType data can be transformed into input for SSAM software. I have some Python code which does this. 

As the results were not so convincing, I focused on SUMO SSM device. A lot of information about the road network has to be provided and current SUMO behaviour complicates things sometimes (e.g. pedestrians waiting on the very sidewalk border counted as "colliding" with turning vehicles).

**Comment by namdre** [comment]
Thanks @m-kro. I misunderstood a question on the mailing list.
Do you think the conversion from fcd to a ssam-compatible would be a relevant addition to tools/traceExporter.py?

**Comment by m-kro** [comment]
I presume not many users would be interested in this feature. Just discovered that the Open Source / Open Data website of the US DOT is currently unavailable  - and so is the SSAM program and the data specs. 
Besides that, it should be easy to create an approximate conversion from FCD output to SSAM trj format.

**Comment by uwmyuan** [comment]
@m-kro Hi, I have a similar problem regarding using the SSAM model with the FCD output of SUMO. FYI, SSAM is an open source software funded by USA FHWA, TRJ is an open-source file format developed by Siemens ITS maintained by New Global Systems for Intelligent Transportation Management, and the documentation of this format is available from the [repo](https://github.com/OSADP/SSAM/blob/master/Documentation/SSAM%20File%20Format%20v1.04.pdf). Could you kindly share your Python script for my personal reference?

@namdre SSAM is a widely used safety evaluation tool. Its input file format, TRJ, is supported by many compatible simulators, such as VISSIM, AIMSUN, Paramics, and TEXAS. An FCD-to-TRJ file converter may help the impact of SUMO. Is that of any interest to add a converter to `tools/` as an enhancement?

**Comment by namdre** [comment]
@uwmyuan I'd be happy to accept pull request for conversion code. Note, that we already have a tool that handles conversion from sumo fcd-output to various trajectory formats (https://sumo.dlr.de/docs/Tools/TraceExporter.html).  Thus, my preferred integration would be via an extra option for that tool and a new converter module in https://github.com/eclipse/sumo/tree/master/tools/sumolib/output/convert

**Comment by m-kro** [comment]
OK, I see what i can do. By the way, just found the necessary[ SSAM documents and software](https://github.com/OSADP/SSAM) are on Github!

**Comment by uwmyuan** [comment]
@namdre Thanks for the information. 

@m-kro Glad you got it. Since you have already done some work on this, would you like to address this issue with the info? Please kindly let me know.

**Comment by m-kro** [comment]
@uwmyuan If you want to test my converter code right away with your test data, please do so and report if everything works well.

**Comment by uwmyuan** [comment]
@m-kro Fantastic, I'll test your code and let you know.

**Comment by uwmyuan** [comment]
I tested the code with the cmd 
```
python "D:\Program Files (x86)\Eclipse\Sumo\tools\traceExporter.py" -i "C:\Users\Yun\p0.5v400_fcd.xml" -n "C:\Users\Yun\osm.net.xml" --trj-output "p0.5v400_trj.trj"
```
and got the following error:
```
Traceback (most recent call last):
  File "D:\Program Files (x86)\Eclipse\Sumo\tools\traceExporter.py", line 373, in <module>
    sys.exit(main(sys.argv))
  File "D:\Program Files (x86)\Eclipse\Sumo\tools\traceExporter.py", line 367, in main
    {"length":options.trjvehlength, "width":options.trjvehwidth, "bbox": net.getBBoxXY(), "timestep":options.timestep})
  File "D:\Program Files (x86)\Eclipse\Sumo\tools\traceExporter.py", line 141, in runMethod
    ret = writer(procFCDStream(fcdStream, options), o, further)
  File "D:\Program Files (x86)\Eclipse\Sumo\tools\sumolib\output\convert\trj.py", line 90, in fcd2trj
    outSTRM.write(struct.pack("=c", bytes(str(laneIndex))))
TypeError: string argument without an encoding
```
It looks related to string encoding. My env is Win10, SUMO1.8.
